integrations/sources/amazon-msk.mdx

```diff
@@ -173,7 +173,7 @@ WITH (
   connector = 'kafka', topic = '<topic-name>',
   properties.bootstrap.server = '<broker-url>',
   scan.startup.mode = 'earliest',
-  properties.sasl.mechanism = 'SCRAM-SHA-512',
+  properties.sasl.mechanism = 'AWS_MSK_IAM',
```

Comment:
This seems problematic to me. The whole tutorial is to configure MSK to use SASL/SCRAM, so we can't simply change it here.

<img width="1234" alt="image" src="https://github.com/user-attachments/assets/95f21a18-27d9-4e13-aa51-bfc1acf1fd3e" />

Besides, to use IAM, we have to give RisingWave the IAM role with permission to access the MSK, and there's no username and password. 

This also means that IAM can only be used for on-prem/BYOC, but not cloud. So perhaps we should keep SCRAM in the doc. Maybe add another section for IAM.

cc @yuhao-su to confirm my understanding.

Reply:
True, it is a glitch indeed.
For IAM, RW loads credentials from EC2 and does not require explicit AK/SK, and yes, for on-perm only.

Reply:
BTW, for SASL/SCRAM, it also seems not possible for cloud, because we need to turn on public access for MSK, which will then require ACL, and we can't add RW cloud to the ACL.

Is it even possible to connect to MSK on cloud? 🤔

Reply:
Yes, the brokers are still available when with the help of private link. 

> public access for MSK

why we need a public access? Or, why assume role cannot work.

Reply:
Oh, indeed private link may help.

Reply:
> why assume role cannot work

Can we assume role in the users' account? 🤔

Reply:
It's a service account that assumes role, and we use that account's AKSK to connect